src/pool/pool_disjoint.c

```diff
@@ -1133,6 +1133,46 @@ static umf_result_t disjoint_pool_get_name(void *pool, const char **name) {
     return UMF_RESULT_SUCCESS;
 }
 
+umf_result_t disjoint_pool_trim_memory(void *pool, size_t minBytesToKeep) {
+    assert(pool != NULL);
+    disjoint_pool_t *hPool = (disjoint_pool_t *)pool;
+
+    // tracking the number of bytes left to keep
+    int bytesLeftToKeep = (int)minBytesToKeep;
```

Comment:
You cast from 8 byte unsigned variable to 4 byte signed variable

Reply:
done